src/tiffcomposite_int.cpp

```diff
@@ -488,7 +488,7 @@ TiffComponent* TiffDirectory::doAddPath(uint16_t tag, TiffPath& tiffPath, TiffCo
       return this->addNext(std::move(atc));
     return this->addChild(std::move(atc));
   }();
-  return tc->addPath(tag, tiffPath, pRoot, std::move(object));
+  return tc->addPath(tag, tiffPath, pRoot, nullptr);
```

Comment:
Why are you applying this change? It does not make sense at first glance 😰

Reply:
So clang-tidy reports here (and it's correct) that there is a double move of object happening, which results in nullptr anyway. This just silences that. Actually both sonarlint and clang-tidy report it.

Reply:
I think clang-tidy is saying that a double move can happen _sometimes_. I don't think it's saying that `object` is _always_ null here. The `std::move` on line 477 is inside an if-statement, so it should be possible to bypass.

Reply:
I don't think adding a null check makes sense honestly.

Reply:
Here's a different suggestion: #2298

Reply:
Applied and removed here.